packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -6,8 +6,8 @@ import {
 } from '@mui/utils';
 import useLazyRef from '@mui/utils/useLazyRef';
 import useTimeout from '@mui/utils/useTimeout';
-import { useResizeObserver } from '@mui/x-internals/useResizeObserver';
 import { useRtl } from '@mui/system/RtlProvider';
+import reactMajor from '@mui/x-internals/reactMajor';
```

Comment:
Just noticed this while working on React 19 bump.
@mui/xgrid Was there a reason to create a local utility, when everywhere else it is imported from `@mui/internal-test-utils`? 🤔

Reply:
We needed this in the data grid package itself, not in the test files.

Reply:
A possible solution https://github.com/mui/base-ui/pull/1047. We could argue that if we can do `import reactMajor from '@mui/x-internals/reactMajor';` then we should remove `import { reactMajor } from '@mui/internal-test-utils';` as it can be a footgun.